### PR TITLE
=str Reduce one field in MapAsyncUnordered.

### PR DESCRIPTION
**Pull request by He-Pin**
I think this can be optimized out to reduce one field.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala [hunk 1]
```diff
@@ -1422,9 +1426,6 @@ private[stream] object Collect {
         }
       }
 
-      private val futureCB = getAsyncCallback(futureCompleted)
```
Comment (He-Pin) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1425]: this field is removed

File: akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala [hunk 2]
```diff
@@ -1397,10 +1397,14 @@ private[stream] object Collect {
 
       private var inFlight = 0
       private var buffer: BufferImpl[Out] = _
+      private var invokeFutureCB: Try[Out] => Unit = _
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1400]: Couldn't we even initialize it right away and make it a val?

Comment (He-Pin) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1400]: As the document said , the `getAsyncCallback` is better to be called from `preStart`，there are some race condition in  the constructor，is that true?

Comment (He-Pin) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1400]: Sharing the AsyncCallback from the constructor risks race conditions, therefore it is recommended to use the preStart() lifecycle hook instead.

Sharing the AsyncCallback from the constructor risks race conditions, therefore it is recommended to use the preStart() lifecycle hook instead.

Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1400]: No sharing of the callback (as in with an outside object at construction) going on here, only calling it triggered by stream push, so it is safe

Comment (He-Pin) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:1400]: good to know that.

**Review by johanandren — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**